pallets/stateful-storage/src/lib.rs

```diff
@@ -290,4 +336,33 @@ pub mod pallet {
 	}
 }
 
-impl<T: Config> Pallet<T> {}
+impl<T: Config> Pallet<T> {
+	fn check_origin(origin: OriginFor<T>) -> Result<MessageSourceId, DispatchError> {
```

Comment:
I see what you did here. Though it seems that ensure_valid_msa_key contributes to checking the origin, they are two different things.

When one checks the origin, it simply checks that the origin enum is a `Signed` origin. It does not do any DB reads. So it would be best not to wrap these together or change the name of this function. I do not have any recommendations for naming at the moment.

Reply:
Gotcha.

Reply:
Refactored.